Added docs for loader options plugin (and placeholder for sourcemap)

johnnyreilly opened the pull request:
This PR adds docs for the loader options plugin (as well as a placeholder for SourceMapDevToolPlugin https://github.com/webpack/webpack.js.org/issues/494  ).

The docs may well be inaccurate - that's fine; set me straight!

johnnyreilly commented on content/plugins/loader-options-plugin.md:
```diff
@@ -0,0 +1,34 @@
+---
+title: loader-options-plugin
+---
```
will do!

bebraw commented on content/plugins/loader-options-plugin.md:
````diff
@@ -0,0 +1,37 @@
+})
+```
+
+contributors:
````
This should go to YAML head matter. The system picks it up from there.

bebraw commented on content/plugins/source-map-dev-tool-plugin.md:
```diff
@@ -0,0 +1,5 @@
+---
+title: source-map-dev-tool-plugin.md
+---
```
Add contributors here too.

bebraw commented on content/plugins/source-map-dev-tool-plugin.md:
```diff
@@ -0,0 +1,5 @@
+title: source-map-dev-tool-plugin.md
+---
+
+?> TODO
```
Easiest way out - pick up the content from https://webpack.github.io/docs/list-of-plugins.html#sourcemapdevtoolplugin source (markdown) and format to fit.

bebraw commented on content/plugins/loader-options-plugin.md:
````diff
@@ -0,0 +1,37 @@
+new webpack.LoaderOptionsPlugin(options)
+```
+
+* `options.debug` (`boolean`): Whether loaders should be in `debug` mode or not.
````
Maybe it's good to mention that the `debug` flag will disappear (I remember reading about that somewhere, correct me if I'm wrong).

bebraw replied:
Yeah, let's wait a comment from Tobias on that.

bebraw commented on content/plugins/loader-options-plugin.md:
````diff
@@ -0,0 +1,37 @@
+```
+
+* `options.debug` (`boolean`): Whether loaders should be in `debug` mode or not.
+* `options.minimize` (`boolean`): Where loaders can be switched to minimize mode.
````
Maybe it's good to expand on what this means and give a concrete example.

johnnyreilly replied:
I don't know what it means - I improvised this.

bebraw replied:
Consider something like *css-loader*. It won't `minimize` by default, but this should do it I think. You can se it per loader, but it might be able to pick up this global config. Worth a quick go.

bebraw commented on content/plugins/loader-options-plugin.md:
```diff
@@ -0,0 +1,37 @@
+
+* `options.debug` (`boolean`): Whether loaders should be in `debug` mode or not.
+* `options.minimize` (`boolean`): Where loaders can be switched to minimize mode.
+* `options.options` (`object`): A configuration object that can be used to configure older loaders - this will take the same schema a `webpack.config.js`
```
Give a couple of examples of older loaders.

johnnyreilly replied:
I don't know which they are; @sokra mentioned it but he didn't specify.  Happy to add it if someone can tell me.

bebraw replied:
*sass-loader*, *less-loader* at least. You would have to look up their readme files to see if they have root level configuration in place as this plugin literally replaces that.

bebraw commented on content/plugins/loader-options-plugin.md:
```diff
@@ -0,0 +1,37 @@
+* `options.minimize` (`boolean`): Where loaders can be switched to minimize mode.
+* `options.options` (`object`): A configuration object that can be used to configure older loaders - this will take the same schema a `webpack.config.js`
+
+* `options.options.context` (`string`): The context that can be used to configure older loaders
```
This needs some example as `context` is too abstract name.

bebraw replied:
Let's wait Tobias on this.

johnnyreilly commented:
I've addressed comments as best I can; as you can see I'm fairly ignorant of the plugins; I know enough to get by.  If the questions asked can be answered by someone I'm happy to update the PR accordingly.

If anything, this is exactly the reason why I think it's valuable to raise this PR; it draws attention to the fact that there are unknowns about webpack plugins that we should turn into docs.  But I can't answer questions I don't know the answer to; this is where you get to tell me what should actually be there!

TheLarkInn commented:
Thank you.

sokra commented on content/plugins/source-map-dev-tool-plugin.md:
```diff
@@ -0,0 +1,26 @@
+---
+title: source-map-dev-tool-plugin.md
```
`.md` is not needed here

johnnyreilly commented:
Is the publication of the docs automated? I can't see the new docs on the site...

bebraw commented:
> Is the publication of the docs automated? I can't see the new docs on the site...

Nope. Waiting for #348 and before that we might want to get the current PRs merged as much as possible.

It would be good to have that title (`title: source-map-dev-tool-plugin.md`) fixed btw.
